[DOC] Kafka connect user access rights (#2852)
strimzi/strimzi-kafka-operator: documentation/modules/proc-configuring-kafka-connect-user-authorization.adoc

// Module included in the following assemblies:
//
// assembly-kafka-connect-configuration.adoc

[id='proc-configuring-kafka-connect-user-authorization-{context}']
= Configuring Kafka Connect user authorization

This procedure describes how to authorize user access to Kafka Connect.

> When simple authorization is being used to access Kafka connect, a Kafka Connect user requires access rights to the consumer group

Reviewer: ```suggestion
When simple authorization is being used in Kafka, a Kafka Connect user requires access rights to the consumer group
```

and the internal topics of Kafka Connect.

> Simple authorization uses ACL rules to provide the right level of access, which are handled by the Kafka `SimpleAclAuthorizer` plugin.

Reviewer: Again, I think we should clarify that this example is for use with `simple` authorization, but that other authorization types will also need to give the rights to the user.

Author: > When **any type** of authorization is being used in Kafka, a Kafka Connect user requires access rights to the consumer group and the internal topics of Kafka Connect.
> 
> In this procedure we show how access is provided when `simple` authorization is being used.

// Module included in the following assemblies:
//
// assembly-kafka-connect-configuration.adoc

[id='proc-configuring-kafka-connect-user-authorization-{context}']
= Configuring Kafka Connect user authorization

This procedure describes how to authorize user access to Kafka Connect.

When any type of authorization is being used in Kafka, a Kafka Connect user requires access rights to the consumer group
and the internal topics of Kafka Connect.

> In this procedure, we show how access is provided when `simple` authorization is being used.

Reviewer 2: ```suggestion
This procedure shows how access is provided when `simple` authorization is being used.
```

> Simple authorization uses ACL rules to provide the right level of access, which are handled by the Kafka `SimpleAclAuthorizer` plugin.

Reviewer 2: ```suggestion
Simple authorization uses ACL rules, handled by the Kafka `SimpleAclAuthorizer` plugin, to provide the right level of access.
```

For more information on configuring a `KafkaUser` resource to use simple authorization, see xref:ref-kafka-user-using-uo[Kafka User resource].

NOTE: The default values for the consumer group and topics will differ when xref:con-kafka-connect-multiple-instances-{context}[running multiple instances].

.Prerequisites

* A Kubernetes cluster
* A running Cluster Operator

.Procedure

. Edit the `authorization` property in the `KafkaUser` resource to provide access rights to the user.
+

> For example, here we see access rights configured for the Kafka Connect topics and consumer group using the following `literal` name values:

Reviewer 2: ```suggestion
For example, here access rights are configured for the Kafka Connect topics and consumer group using the following `literal` name values:
```

. Edit the `authorization` property in the `KafkaUser` resource to provide access rights to the user.
+

> For example, here access rights are configured for the Kafka Connect topics and consumer group using the following `literal` name values:

Reviewer 2: This reads a bit awkwardly now (sorry!). Why not say something like:

"In the following example, access rights are configured..."
